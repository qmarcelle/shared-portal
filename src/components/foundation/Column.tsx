import { ReactNode } from 'react';
import { IComponent } from '../IComponent';

interface ColumnProps extends IComponent {
  children: ReactNode;
}

export const Column = ({
  className = '',
  children,
  onClick,
  tabIndex,
}: ColumnProps) => {
  return (
    <div
      tabIndex={tabIndex}
      className={`flex flex-col ${className}`.trimEnd()}
      onClick={onClick}
    >
      {children}
    </div>
  );
};
