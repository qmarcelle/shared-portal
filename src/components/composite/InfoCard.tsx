import Image from 'next/image';
import { IComponent } from '../IComponent';
import { Card } from '../foundation/Card';
import { Column } from '../foundation/Column';
import { Row } from '../foundation/Row';
import { TextBox } from '../foundation/TextBox';

interface InfoCardProps extends IComponent {
  label: string;
  body: string;
  icon: string;
  link?: string;
}

export const InfoCard = ({ label, body, icon, link }: InfoCardProps) => {
  return (
    <a href={link}>
      <Card type="elevated" key={label} className="small-section">
        <Row>
          <Image className="size-10" src={icon} alt="link" />
          <Column className="ml-4">
            <TextBox className="link-row-head" text={label} />
            <TextBox className="body-1" text={body} />
          </Column>
        </Row>
      </Card>
    </a>
  );
};
