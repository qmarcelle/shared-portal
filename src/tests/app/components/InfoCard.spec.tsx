import { InfoCard } from '@/components/composite/InfoCard';
import { Column } from '@/components/foundation/Column';
import {
  benefitsCoverage,
  claimsBenefitsCoverage,
  planDocuments,
  priorAuthorizations,
  spendingAccounts,
  spendingSummary,
} from '@/components/foundation/Icons';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const ViewOtherPlanInformationDetails = [
  {
    label: 'Benefits & Coverage',
    description: 'View what&apos;s covered under your plan.',
    iconName: benefitsCoverage,
    link: '/benefits',
  },
  {
    label: 'Claims',
    description: 'Search for claims and view details or submit a claim.',
    iconName: claimsBenefitsCoverage,
    link: '/claims',
  },
  {
    label: 'Prior Authorizations',
    description: 'Check the status of your prior authorizations.',
    iconName: priorAuthorizations,
    link: '/priorAuth',
  },
  {
    label: 'Spending Accounts',
    description: 'Check your HSA, HRA or FSA accounts.',
    iconName: spendingAccounts,
    link: '/spendingAccount',
  },
  {
    label: 'Spending Summary',
    description:
      'Your annual statement shows claims we received and processed.',
    iconName: spendingSummary,
    link: '/spendingSummary',
  },
  {
    label: 'Plan Documents',
    description: 'Your important plan documents.',
    iconName: planDocuments,
    link: '/planDocuments',
  },
];

const renderUI = () => {
  return render(
    <Column>
      {ViewOtherPlanInformationDetails.map((item) => {
        return (
          <InfoCard
            key={item.label}
            label={item.label}
            icon={item.iconName}
            body={item.description}
            link={item.link}
          />
        );
      })}
    </Column>,
  );
};

describe('View Other Plan Information', () => {
  it('should render UI correctly', () => {
    const component = renderUI();

    screen.getByText('Benefits & Coverage');
    screen.getByText('View what&apos;s covered under your plan.');
    expect(component.baseElement).toMatchSnapshot();

    screen.getByText('Claims');
    screen.getByText('Search for claims and view details or submit a claim.');
    expect(component.baseElement).toMatchSnapshot();

    screen.getByText('Prior Authorizations');
    screen.getByText('Check the status of your prior authorizations.');
    expect(component.baseElement).toMatchSnapshot();

    screen.getByText('Spending Accounts');
    screen.getByText('Check your HSA, HRA or FSA accounts.');
    expect(component.baseElement).toMatchSnapshot();

    screen.getByText('Spending Summary');
    screen.getByText(
      'Your annual statement shows claims we received and processed.',
    );
    expect(component.baseElement).toMatchSnapshot();

    screen.getByText('Plan Documents');
    screen.getByText('Your important plan documents.');
    expect(component.baseElement).toMatchSnapshot();
  });
});
