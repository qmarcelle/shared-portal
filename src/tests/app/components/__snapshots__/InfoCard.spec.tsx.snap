// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Other Plan Information should render UI correctly 1`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/benefits"
            >
              <p
                class="body-1 link-row-head"
              >
                Benefits & Coverage
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              View what's covered under your plan.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/claims"
            >
              <p
                class="body-1 link-row-head"
              >
                Claims
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Search for claims and view details or submit a claim.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/priorAuth"
            >
              <p
                class="body-1 link-row-head"
              >
                Prior Authorizations
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check the status of your prior authorizations.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingAccount"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Accounts
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check your HSA, HRA or FSA accounts.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingSummary"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Summary
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your annual statement shows claims we received and processed.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/planDocuments"
            >
              <p
                class="body-1 link-row-head"
              >
                Plan Documents
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your important plan documents.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`View Other Plan Information should render UI correctly 2`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/benefits"
            >
              <p
                class="body-1 link-row-head"
              >
                Benefits & Coverage
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              View what's covered under your plan.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/claims"
            >
              <p
                class="body-1 link-row-head"
              >
                Claims
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Search for claims and view details or submit a claim.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/priorAuth"
            >
              <p
                class="body-1 link-row-head"
              >
                Prior Authorizations
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check the status of your prior authorizations.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingAccount"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Accounts
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check your HSA, HRA or FSA accounts.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingSummary"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Summary
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your annual statement shows claims we received and processed.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/planDocuments"
            >
              <p
                class="body-1 link-row-head"
              >
                Plan Documents
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your important plan documents.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`View Other Plan Information should render UI correctly 3`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/benefits"
            >
              <p
                class="body-1 link-row-head"
              >
                Benefits & Coverage
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              View what's covered under your plan.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/claims"
            >
              <p
                class="body-1 link-row-head"
              >
                Claims
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Search for claims and view details or submit a claim.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/priorAuth"
            >
              <p
                class="body-1 link-row-head"
              >
                Prior Authorizations
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check the status of your prior authorizations.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingAccount"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Accounts
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check your HSA, HRA or FSA accounts.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingSummary"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Summary
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your annual statement shows claims we received and processed.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/planDocuments"
            >
              <p
                class="body-1 link-row-head"
              >
                Plan Documents
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your important plan documents.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`View Other Plan Information should render UI correctly 4`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/benefits"
            >
              <p
                class="body-1 link-row-head"
              >
                Benefits & Coverage
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              View what's covered under your plan.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/claims"
            >
              <p
                class="body-1 link-row-head"
              >
                Claims
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Search for claims and view details or submit a claim.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/priorAuth"
            >
              <p
                class="body-1 link-row-head"
              >
                Prior Authorizations
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check the status of your prior authorizations.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingAccount"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Accounts
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check your HSA, HRA or FSA accounts.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingSummary"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Summary
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your annual statement shows claims we received and processed.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/planDocuments"
            >
              <p
                class="body-1 link-row-head"
              >
                Plan Documents
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your important plan documents.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`View Other Plan Information should render UI correctly 5`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/benefits"
            >
              <p
                class="body-1 link-row-head"
              >
                Benefits & Coverage
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              View what's covered under your plan.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/claims"
            >
              <p
                class="body-1 link-row-head"
              >
                Claims
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Search for claims and view details or submit a claim.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/priorAuth"
            >
              <p
                class="body-1 link-row-head"
              >
                Prior Authorizations
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check the status of your prior authorizations.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingAccount"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Accounts
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check your HSA, HRA or FSA accounts.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingSummary"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Summary
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your annual statement shows claims we received and processed.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/planDocuments"
            >
              <p
                class="body-1 link-row-head"
              >
                Plan Documents
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your important plan documents.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`View Other Plan Information should render UI correctly 6`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/benefits"
            >
              <p
                class="body-1 link-row-head"
              >
                Benefits & Coverage
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              View what's covered under your plan.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/claims"
            >
              <p
                class="body-1 link-row-head"
              >
                Claims
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Search for claims and view details or submit a claim.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/priorAuth"
            >
              <p
                class="body-1 link-row-head"
              >
                Prior Authorizations
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check the status of your prior authorizations.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingAccount"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Accounts
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Check your HSA, HRA or FSA accounts.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/spendingSummary"
            >
              <p
                class="body-1 link-row-head"
              >
                Spending Summary
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your annual statement shows claims we received and processed.
            </p>
          </div>
        </div>
      </div>
      <div
        class="card-elevated small-section"
      >
        <div
          class="flex flex-row"
        >
          <img
            alt="link"
            class="size-10"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="flex flex-col ml-4"
          >
            <a
              href="/planDocuments"
            >
              <p
                class="body-1 link-row-head"
              >
                Plan Documents
              </p>
            </a>
            <p
              class="body-1 body-1"
            >
              Your important plan documents.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
